virtual/environment_test.go

```diff
@@ -688,6 +688,134 @@ func testHeartbeatAndRebalancingWithMemory(
 	}
 }
 
+// TestReplicationMemoryLoadBalancingGoModule tests the replication logic of the environment with respect to memory load balancing.
+//
+// This test specifically examines how actors are replicated when the ExtraReplicas option is set to a value greater than 0.
+// The test verifies that the actor is replicated if the memory usage of the server where the actor is currently activated is not the lowest among available servers,
+// and there are other servers with lower memory usage that can accommodate additional replicas.
+func TestReplicationMemoryLoadBalancingGoModule(t *testing.T) {
```

Comment:
I dont think the behavior you're testing here makes a lot of sense (since all replicas should use ~= memory usage). I would change this to do something much simpler like just make sure we're ~ evenly splitting the requests between both replicas, or that we're sending requests to both replicas at all.